Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutVaultRequest.swift

```diff
@@ -0,0 +1,59 @@
+#if canImport(BraintreePayPal)
+import BraintreePayPal
+#endif
+
+/**
+ Options for the PayPal Vault flow.
+ */
+@objc public class BTPayPalNativeVaultRequest: BTPayPalVaultRequest, BTPayPalNativeRequest {
+
+    let hermesPath: String
+
+    let paymentType: BTPayPalPaymentType = .vault
+
+    /**
+     Initializes a PayPal Vault request.
+
+     - Returns: A PayPal Vault request.
+     */
+    @objc public override init() {
+        self.hermesPath = "v1/paypal_hermes/setup_billing_agreement"
+
+        super.init()
+    }
+
+    func parameters(with configuration: BTConfiguration) -> [AnyHashable : Any] {
+
+        let baseParams = getBaseParameters(with: configuration)
+
+        // Should only include shipping params if they exist
+        let shippingParams: [AnyHashable: Any?]? = {
+            if let shippingOverride = shippingAddressOverride {
+                return [
+                  "line1": shippingOverride.streetAddress,
+                  "line2": shippingOverride.extendedAddress,
+                  "city": shippingOverride.locality,
+                  "state": shippingOverride.region,
+                  "postal_code": shippingOverride.postalCode,
+                  "country_code": shippingOverride.countryCodeAlpha2,
+                  "recipient_name": shippingOverride.recipientName,
+                ]
+            }
+            else {
```

Comment:
```suggestion
            } else {
```

Reply:
We don't currently but it is on our todo and will likely be similar to the Unified SDK's linter rules